robot-core/src/main/java/org/obolibrary/robot/ReportOperation.java

```diff
@@ -695,20 +699,21 @@ private static Map<String, String> getDefaultQueryStrings(Set<String> rules)
     Map<String, String> queries = new HashMap<>();
     // Handle simple file path, probably accessed during testing
     if (dirURL != null && dirURL.getProtocol().equals("file")) {
-      String[] queryFilePaths = new File(dirURL.toURI()).list();
-      if (queryFilePaths == null || queryFilePaths.length == 0) {
+      final Set<Path> queryFilePaths =
+          Files.list(Paths.get(dirURL.toURI())).collect(Collectors.toSet());
+      if (queryFilePaths.size() == 0) {
         throw new IOException(
             "Cannot access report query files. There are no files in the directory.");
       }
       Set<String> defaultRuleNames = new HashSet<>();
-      for (String qPath : queryFilePaths) {
-        String ruleName = qPath.substring(qPath.lastIndexOf("/")).split(".")[0];
+      for (final Path qPath : queryFilePaths) {
+        final String ruleName = qPath.getFileName().toString().split("[.]")[0];
```

Comment:
These kinds of changes are very hard to review and we prefer in the interest of resources to only do them if we have an issue that cleary explains what is wrong, with agreement all around it absolutely needs changing and then changing it. Are they _absolutely essential_ to this PR?

Reply:
Thanks @matentzn for asking to keep the PR focused, and sorry if I’m being inconsistent, but this path handling looks like a nice improvement.

Reply:
@matentzn you are absolutely right. Actually, I originally wanted to "hang" the test on the existing API of ReportOperation - the changes were needed to make it actually working in tests.

But this would have mean going through profile.txt which would be an extra overhead so I reconsidered and used the getViolations method to keep it focused.

This bit of code belonged to the original story and I accidentally committed. I reverted it back. I also don't like these types of extra changes - doing unnecessary changes when there are almost no tests around is dangerous.